### ISSUE TITLE
Unexpected behaviour when using credo diff

### ISSUE THREAD

**Issue by alanvardy** [rrrene/credo#829]
### Environment

* Credo version (`mix credo -v`): 1.5.1
* Erlang/Elixir version (`elixir -v`):
Erlang/OTP 22 [erts-10.7] [source] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [hipe]
Elixir 1.10.4 (compiled with Erlang/OTP 22)

* Operating system:
Pop_os 20.10 (Ubuntu derivative)

### What were you trying to do?
Run `mix credo diff master -C stricter` on either `master` or a branch off of `master` where the only difference between master and head is the change to .credo.exs adding the "stricter" config.

Stricter has all checks disabled except for `Credo.Check.Readability.ModuleDoc`

### Expected outcome

No errors, because no change has been made to code files

### Actual outcome

```bash
❯ mix credo diff master -C stricter
Diffing 1150 source files in working dir with master (this might take a while) ...

    Code Readability                                                                                                                                      
  ┃ 
+ ┃ [R] → Modules should have a @moduledoc tag.
+ ┃       apps/some_app/lib/some_app/context/is_expired.ex:1:11 #(SomeApp.Context.IsExpired)
+ ┃ [R] → Modules should have a @moduledoc tag.
+ ┃       apps/some_app/lib/some_app/context/is_boolean.ex:1:11 #(SomeApp.Context.IsBoolean)
+ ┃ [R] → Modules should have a @moduledoc tag.
+ ┃       apps/some_app/lib/some_app/context/exists.ex:1:11 #(SomeApp.Context.Exists)
+ ┃ [R] → Modules should have a @moduledoc tag.
+ ┃       apps/some_app/lib/some_app/context/constants.ex:1:11 #(SomeApp.Context.Constants)
+ ┃ [R] → Modules should have a @moduledoc tag.
+ ┃       apps/some_app/lib/some_app/config.ex:1:11 #(SomeApp.Config)

... and more

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.5 seconds (0.2s to load, 0.3s running 2 checks on 1150 files)

Changes between master and working dir:

+  added 547 new code readability issues,
✔  fixed no issues, and
~  kept no issues.

Use `mix credo explain` to explain issues, `mix credo diff --help` for options.
```

Results are identitical whether on master with changes to .credo.exs uncommitted, or on a branch with changes to .credo.exs committed.

Am I performing the command correctly?

For context: this command was run on an umbrella project, I don't if that changes anything.

Thank you for all that you do.

**Comment by rrrene** [comment]
@alanvardy You're right. I was able to reproduce this, something is not right here. Thx for reporting :+1:

**Comment by rrrene** [comment]
My first impression is this:

I did not have a config named `strict` on `master` when I tested this. So what happened behind the scenes is that Credo running on `master` with a none-existing config resulted in no issues. Credo running in the working dir (with existing config `stricter`) resulted in the issues on display.

So, while Credo is "right" from its point of view, this is clear nonsense. Credo is doing *something* wrong here, but we have to decide what the (mostly) expected behaviour would be.

What should happen when the "before" `.credo.exs` does not have the config you specify? Should it use `default`? WDYT?

**Comment by alanvardy** [comment]
Oh, I didn't know that it was the error messages being diffed between old and new rather than a smaller file list being diffed by the latest config.

My use case is that I am working on a large codebase where I would like to introduce new new lints gradually in CI using a "you touched so now you own it" kind of mentality where once someone makes a change in a module that module is now held to the new stricter standard.

I am hoping to add new lints to the "stricter" config and let such things as moduledocs gradually be added over time.

Another problem now that I think about it is that if I were to add a lint in the future to the "stricter" config, it would also raise complaints about that one across the whole code base as the old stricter wouldn't have the lint enabled while the new one would. Am I correct in that assumption?

Perhaps the only way to do this is to get a list of files that changed between the specified commit and the current commit, and then run the latest config on that list of files. Code consistency checks may need to be disabled for this as they wouldn't be representative of the whole system. Maybe this is a different feature than diff?

To directly reply to what diff should do in the case of not having the same config in both commits, I suspect it may not be reconcilable. Perhaps an error message? `"Config #{config} not available on branch #{branch}"`

**Comment by rrrene** [comment]
Update: There's a "deeper" problem in Credo's handling of "missing" configs. I will try to release a fix for this soon. Thx for reporting this, it let me down a rabbit hole worth exploring :+1:

**Comment by rrrene** [comment]
@alanvardy Could you try setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

It should now give you an error message. This is not the final fix for the overall problem you uncovered, but it is a bandaid I would want to release soon.

**Comment by alanvardy** [comment]
Cool. I got this message:

```elixir
** (diff) Running Credo on `master` (checked out to /tmp/credo-diff-20201206232519572718) resulted in the following error:

Given config name "stricter" does not exist in any config file:

  * /tmp/credo-diff-20201206232519572718/.credo.exs

warning: Execution halted during Elixir.Credo.Execution.Task.RunCommand!
  lib/credo/execution/task/run_command.ex:4: Credo.Execution.Task.RunCommand.error/2
  lib/credo/execution/task.ex:60: Credo.Execution.Task.do_run/3
  (elixir 1.10.4) lib/enum.ex:2111: Enum."-reduce/3-lists^foldl/2-0-"/3
  lib/credo.ex:30: Credo.run/1
  lib/credo/cli.ex:41: Credo.CLI.run_to_halt/1
```

**Comment by rrrene** [comment]
Forgot to close the issue: `v1.5.4` includes this fix 👍
